sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -1350,6 +1351,42 @@ export class BlobClient extends StorageClient {
     }
   }
 
+  /**
+   * Marks the specified blob or snapshot for deletion if it exists. The blob is later deleted
+   * during garbage collection. Note that in order to delete a blob, you must delete
+   * all of its snapshots. You can delete both at the same time with the Delete
+   * Blob operation.
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob
+   *
+   * @param {BlobDeleteOptions} [options] Optional options to Blob Delete operation.
+   * @returns {Promise<BlobDeleteResponse | null>} Returns null if the blob/snapshot does not exist.
+   * @memberof BlobClient
+   */
+  public async deleteIfExists(options: BlobDeleteOptions = {}): Promise<BlobDeleteResponse | null> {
+    const { span, spanOptions } = createSpan("BlobClient-deleteIfExists", options.tracingOptions);
+    try {
+      return await this.delete({
+        ...options,
+        tracingOptions: { ...options!.tracingOptions, spanOptions }
+      });
+    } catch (e) {
+      if (e.details?.errorCode === "BlobNotFound") {
```

Comment:
Please double check if errorCode is BlobNotFound in snapshot case. And what about blob versioning case?

Reply:
Checked. It's the same. Also added unit test code for it.